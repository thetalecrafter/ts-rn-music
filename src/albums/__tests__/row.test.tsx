import { shallow } from 'enzyme'
import * as React from 'react'
import Album from '../album'
import Row from '../row'

describe('Album row', () => {
  const album: Album = {
    artistId: 80204262,
    artistName: 'Various Artists',
    artworkUrl100: 'http://is5.mzstatic.com/image/thumb/Music7/v4/4b/41/61/4b4161fd-4709-60d3-dedd-5b978ba5a2d9/source/100x100bb.jpg',
    artworkUrl60: 'http://is5.mzstatic.com/image/thumb/Music7/v4/4b/41/61/4b4161fd-4709-60d3-dedd-5b978ba5a2d9/source/60x60bb.jpg',
    collectionCensoredName: 'Over the Hedge (Music from the Motion Picture)',
    collectionExplicitness: 'notExplicit',
    collectionId: 1024117649,
    collectionName: 'Over the Hedge (Music from the Motion Picture)',
    collectionPrice: 9.99,
    collectionType: 'Album',
    collectionViewUrl: 'https://itunes.apple.com/us/album/over-the-hedge-music-from-the-motion-picture/1024117649?uo=4',
    copyright: 'â„— 2006 DreamWorks Animation L.L.C., / (P) 2001, 2006 Epic Records, a division of Sony Music Entertainment',
    country: 'USA',
    currency: 'USD',
    primaryGenreName: 'Pop',
    releaseDate: new Date('2006-05-16T07:00:00.000Z'),
    trackCount: 13,
    wrapperType: 'collection'
  }

  it('renders the album image and censored name', () => {
    const tree = shallow(<Row album={album} onPress={jest.fn()} />)
    expect(tree).toMatchSnapshot()
  })

  it('calls back with the album when pressed', () => {
    const onPress = jest.fn()
    const tree = shallow(<Row album={album} onPress={onPress} />)
    tree.find('TouchableHighlight').simulate('Press')
    expect(onPress).toHaveBeenCalledWith(album)
  })
})
