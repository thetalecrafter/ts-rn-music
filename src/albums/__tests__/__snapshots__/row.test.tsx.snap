// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Album row renders the album image and censored name 1`] = `
<TouchableHighlight
  activeOpacity={0.85}
  onPress={[Function]}
  underlayColor="transparent"
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "borderBottomColor": "#C7CDD1",
        "borderBottomWidth": 0.5,
        "flexDirection": "row",
        "minHeight": 54,
        "paddingHorizontal": 16,
        "paddingVertical": 8,
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "http://is5.mzstatic.com/image/thumb/Music7/v4/4b/41/61/4b4161fd-4709-60d3-dedd-5b978ba5a2d9/source/60x60bb.jpg",
        }
      }
      style={
        Object {
          "height": 32,
          "marginRight": 8,
          "width": 32,
        }
      }
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "flex": 1,
        }
      }
    >
      Over the Hedge (Music from the Motion Picture)
    </Text>
  </View>
</TouchableHighlight>
`;
